templates/components/common/navigation-menu.html

```diff
@@ -2,231 +2,96 @@
     <div class="u-hidden@medium">
         {{> components/common/quick-search name='nav-menu-quick-search'}}
     </div>
-    <ul class="js-nav-pages-list{{#if theme_settings.navigation_design '!==' 'simple'}} js-nav-pages-list-depth-max{{/if}}">
+    <ul class="js-nav-pages-list{{#if theme_settings.navigation_design '!==' 'simple'}} js-nav-pages-list-depth-max{{/if}} c-navigation-menu">
         {{#each categories}}
-            <li>
-                {{#if ../theme_settings.navigation_design '===' 'simple'}}
-                    {{> components/common/navigation-list}}
-                {{/if}}
-                {{#if ../theme_settings.navigation_design '===' 'alternate'}}
-                    {{> components/common/navigation-list-alternate}}
-                {{/if}}
-            </li>
+            {{#unless name '===' 'Shop All'}}
+                <li class="c-navigation-menu-item">
+                    {{#if ../theme_settings.navigation_design '===' 'simple'}}
+                        {{> components/common/navigation-list}}
+                    {{/if}}
+                    {{#if ../theme_settings.navigation_design '===' 'alternate'}}
+                        {{> components/common/navigation-list-custom categories=../categories social_media=../social_media }}
+                    {{/if}}
+                </li>
+                <svg class="c-navigation__separator" xmlns="http://www.w3.org/2000/svg" width="25" height="24" viewBox="0 0 25 24" fill="none">
+                    <path d="M13.5 12C13.5 12.5523 13.0523 13 12.5 13C11.9477 13 11.5 12.5523 11.5 12C11.5 11.4477 11.9477 11 12.5 11C13.0523 11 13.5 11.4477 13.5 12Z" stroke="#E6DABB" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
+                </svg>
+            {{/unless}}
         {{/each}}
-        {{#unless theme_settings.hide_content_navigation}}
-            {{#each pages}}
+    </ul>
+    <div class="c-navigation__megamenu is-mobile">
+        <div class="js-quick-search">
+            {{> components/common/quick-search name='nav-menu-quick-search'}}
+        </div>
+        <ul class="c-navigation-list__submenu">
+            {{#each categories}}
+                {{#each children}}
+                    <li>
+                        {{#if children}}
+                            <a 
+                                class="js-nav-pages-action has-sub-menu{{#if is_active}} is-active{{/if}} c-navigation__link c-navigation-list__submenu-header"
```

Comment:
Not a fan of using different component classnames for the same element here - `c-nagivation__link` and `c-nagivation-list__submenu-header` and neither of them uses component's classname which is `navigation-menu` 🤔  If it's possible could it be simplified?